--- /dev/null	2013-04-07 17:22:00.000000000 +0200
+++ cycle.c	2013-04-07 17:22:58.000000000 +0200
@@ -0,0 +1,22 @@
+void nextlayout (const Arg * arg)
+{
+	Layout * l;
+
+	for (l = (Layout *) layouts; l != selmon->lt [selmon->sellt]; l++);
+	if (l->symbol && (l + 1)->symbol)
+		setlayout (&((Arg) { .v = (l+1) }));
+
+	else
+		setlayout (&((Arg) { .v = layouts }));
+}
+
+void prevlayout (const Arg * arg)
+{
+	Layout * l;
+	for (l = (Layout *) layouts; l != selmon->lt[selmon->sellt];l++);
+	if (l != layouts && (l - 1)->symbol)
+		setlayout (&((Arg) { .v = (l-1) }));
+	else
+		setlayout (&((Arg) { .v = &layouts[LENGTH(layouts) - 2] }));
+}
+
